### PR TITLE
Add ability to use xfail and skip markers

### PR DESCRIPTION
**Pull request by ssbarnea**
Allow users to use `xfail` or `skip` markers inside molecule scenario files to modify their execution.

### CODE REVIEW

File: README.rst [hunk 1]
```diff
@@ -56,6 +56,18 @@ This plugin also adds a new pytest option named
...
 
+Using pytest markers
+--------------------
+
+If you want to add various pytest markers like ``xfail`` or ``xskip``,
+just add something like below to your ``molecule.yml`` file:
+
+.. code-block:: yaml
+
+    tags:
```
Comment (tadeboro) [inline README.rst]: Is this name appropriate? I have a feeling we might confuse users here since Ansible also uses tags and those two concepts are not related at all. Maybe we should go with `marks` since this is the name pytest uses?

File: src/pytest_molecule/__init__.py [hunk 2]
```diff
@@ -173,6 +173,13 @@ def __init__(self, name, parent):
             self.molecule_driver = data.get("driver", {}).get("name", "no_driver")
             self.add_marker(self.molecule_driver)
 
+            # check for tags and add them as markers
+            for tag in data.get("tags", []):
+                try:
+                    self.add_marker(tag)
+                except pytest.PytestUnknownMarkWarning:
+                    pass
```
Comment (tadeboro) [inline src/pytest_molecule/__init__.py]: I am not sure how I feel about silently ignoring invalid marks. What is the use case for ignoring invalid ones? I would expect pytest-molecule users to take care of their testing setup and thus should be notified when something went awry.

**Review by tadeboro — DISMISSED** [review]

File: src/pytest_molecule/__init__.py (hunk 2 above)
Comment (ssbarnea) [reply src/pytest_molecule/__init__.py]: I used tags on purpose in order to match ansible as I expect to make use of that for other purposes in the future.

We ignore unknown tags because pylint complains about them. In fact my impression is that they may even become failures in the future. Those that want to use custom marks they need to document them in pytest config file, with description.

File: README.rst (hunk 1 above)
Comment (webknjaz) [reply README.rst]: +1 markers sounds more appropriate. Otherwise, this will cause a lot of confusion.

File: src/pytest_molecule/__init__.py (hunk 2 above)
Comment (webknjaz) [reply src/pytest_molecule/__init__.py]: We shouldn't ignore such markers. Pytest has a built-in mechanism for turning them into errors with `--strict-markers` and it must remain supported for these too.

File: README.rst [hunk 3]
```diff
@@ -56,6 +56,18 @@ This plugin also adds a new pytest option named
 when molecule drivers are not loading. Current default is ``None`` but you
 can choose marks like ``skip`` or ``xfail``.
 
+Using pytest markers
+--------------------
+
+If you want to add various pytest markers like ``xfail`` or ``xskip``,
+just add something like below to your ``molecule.yml`` file:
```
Comment (webknjaz) [inline README.rst]: Semantically, it's better to use the file role here
```suggestion
just add something like below to your :file:`molecule.yml` file:
```

File: src/pytest_molecule/__init__.py [hunk 4]
```diff
@@ -173,6 +173,13 @@ def __init__(self, name, parent):
             self.molecule_driver = data.get("driver", {}).get("name", "no_driver")
             self.add_marker(self.molecule_driver)
 
+            # check for tags and add them as markers
+            for tag in data.get("tags", []):
+                try:
+                    self.add_marker(tag)
```
Comment (webknjaz) [inline src/pytest_molecule/__init__.py]: Some markers have arguments, like conditionals and behavior modifiers. It looks like adding them unconditionally doesn't account for that.

File: README.rst [hunk 5]
```diff
@@ -56,6 +56,18 @@ This plugin also adds a new pytest option named
 when molecule drivers are not loading. Current default is ``None`` but you
 can choose marks like ``skip`` or ``xfail``.
 
+Using pytest markers
+--------------------
+
+If you want to add various pytest markers like ``xfail`` or ``xskip``,
```
Comment (webknjaz) [inline README.rst]:
```suggestion
If you want to add various pytest markers like ``xfail`` or ``skipif``,
```
There's no `xskip`, did you mean `skipif`?

File: src/pytest_molecule/__init__.py (hunk 2 above)
Comment (ssbarnea) [reply src/pytest_molecule/__init__.py]: If we are to not ignore unknown ones, we should add something like `pytest-markers: []` instead of generic `tags: []`, which we could reuse for other features.

My personal take on this is to go for multi-purpose tags/labels like approach, similar to ansible one, where special values have meaning but were you can still use arbitrary values without having to worry that it will break code.

Still, if most are against this approach, I will not try to force my opinion on them.

Probably you observed that the goal was to allow use of `xfail` and `xskip`.

File: src/pytest_molecule/__init__.py (hunk 4 above)
Comment (ssbarnea) [reply src/pytest_molecule/__init__.py]: That is outside the scope of this change. I have no plans to add support for conditionals, not myself. If someone does I will review but I do not find it useful enough. Ability to temporarily mark a test as allowed to fail is enough for me.

File: src/pytest_molecule/__init__.py [hunk 6]
```diff
@@ -173,6 +173,19 @@ def __init__(self, name, parent):
             self.molecule_driver = data.get("driver", {}).get("name", "no_driver")
             self.add_marker(self.molecule_driver)
 
+            # check for known markers and add them
+            markers = data.get("markers", [])
+            if "xfail" in markers:
```
Comment (webknjaz) [inline src/pytest_molecule/__init__.py:178]: Why do you need this check?

Comment (ssbarnea) [reply src/pytest_molecule/__init__.py:178]: I do not understand the question. `data` contains the `molecule.yml` data and we want to add the marker only if the `markers` section is present and if it contains one of them.
